reframe/core/decorators.py

```diff
@@ -262,6 +262,7 @@ def __init__(self):
        Add the ability to define post-init hooks in tests.
 
     '''
+    stage = stage if stage != 'init' else '__init__'
```

Comment:
I will expand this to the other special cases, such as `@run_after('compile')` to create a `post_compile_wait` hook.